### DISCUSSION TITLE
Idea: Scrolling Marquee

### DISCUSSION THREAD

**Discussion by yuuzhan** [Anders-Holst/xled_plus#20] [Ideas]
Hey!  I just found our repository yesterday and looking good so far!  It might be here already, but I was wondering if you had a sample on how to get a scrolling marquee working?  Im guessing this one would work best on 2D layouts :)

**Comment by Anders-Holst** [comment]
And with "marquee" you mean like a rolling sequence of colored bands over the leds?
Then check out the samples day11, day12, and candycane (all in the directory xled_plus/samples).
You can copy them and modify to your own needs. Yes, they assume a 2D layout to work best.

The examples above are based on the effects defined in xled_plus/sequences.py. If you don't want gradients but "hard" borders between colors you can use ColorSequence instead of GradientSequence.

**Reply by yuuzhan** [reply]
Thank you @Anders-Holst for the fast reply!  Scrolling Marquee I meant scrolling text sorry.

I did figure out how to do it with a small modification to show_text.py / RunningText (I am a programmer, but I dont know Python so it took me a bit longer to get there then I care to admit lol)

```
class RunningText(MovingShapesScene):
    def __init__(self, ctr, txt, color, linewidth=0.1, size=0.6, speed=1.0, proj2D3D="cylshell"):
        super(RunningText, self).__init__(ctr)
        self.textcolor = color
        self.lw = linewidth
        self.txt = txt
        self.size = size
        self.speed = speed
        self.horizon = 0
        self.proj2D3D = proj2D3D
        self.place_text()
        self.preferred_frames = self.nsteps
```

Specifically I added a parameter to have proj2D3D not set as a cylinder, which would let it show on a 2d wall just fine.  Before I did that the lights kind of just twinkled seemingly randomly :P

`eff = RunningText(ctr, txt.upper(), makecolorlist(len(txt)), size=.4, speed=.5, linewidth=.2, proj2D3D="")`

Again thank you so much for all your work!
